impl/maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -877,16 +879,11 @@ public void setArtifacts(Set<Artifact> artifacts) {
      */
     public Set<Artifact> getArtifacts() {
```

Comment:
```suggestion
    public synchronized Set<Artifact> getArtifacts() {
```
is it related to concurrency? The `artifacts` referenced in `L:886` should not be `null`.

It it possible that some setter call happens, between `L:883` init and access on `NPE` throwing line?